### PR TITLE
I upgraded react-native to 0.48.1 and fixed the issues. I upgraded th…

### PR DESCRIPTION
**Pull request by MagicPoulp**
…e other packages without seeing any issue.

I could see the change works fine.

upgrading package versions made it work better with the latest Android SDK
Tools like emulator and adb are used in the background by react-native. So upgrading is good.

I fixed the dependency problem for react-test-renderer
I fixed the PropTypes warning when runnning the tests

### CODE REVIEW

**Comment by MagicPoulp** [comment]
By the way, it is a nice example repo. Thanks!

**Comment by MagicPoulp** [comment]
In case you did not know, I found your example at:
http://www.awesome-react-native.com/

here it says:
" Redux Demo ★181 - Minimal implement of redux counter example on ReactNative iOS and Android "

**Comment by MagicPoulp** [comment]
Actually, I only tested that it works for Android. I did not test for iOS.

**Comment by MagicPoulp** [comment]
I updated the first commit with the fixes for both Android and iOS.

File: package.json [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
+  "devDependencies": {
+    "babel-jest": "21.0.0",
+    "babel-preset-react-native": "3.0.2",
+    "jest": "21.0.1",
+    "jest-react-native": "18.0.0",
+    "react-test-renderer": "16.0.0-alpha.12"
+  },
+  "jest": {
+    "preset": "react-native"
+  }
```
Comment (chentsulin) [inline package.json]: A comma is missing here.

File: package.json [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
+    "react-redux": "^5.0.6",
+    "redux": "^3.6.0",
+    "redux-thunk": "^2.1.0",
+    "remote-redux-devtools": "^0.5.5"
+  },
+  "devDependencies": {
+    "babel-jest": "21.0.0",
+    "babel-preset-react-native": "3.0.2",
+    "jest": "21.0.1",
+    "jest-react-native": "18.0.0",
```
Comment (chentsulin) [inline package.json]: It seems that `jest-react-native`can be removed.
https://facebook.github.io/jest/docs/en/tutorial-react-native.html#setup

**Review by chentsulin — CHANGES_REQUESTED** [review]

**Comment by chentsulin** [comment]
@MagicPoulp It looks great, and I will merge this when `package.json` be fixed. Thanks for your help upgrading this repo!

File: package.json (hunk 2 above)
Comment (MagicPoulp) [reply package.json]: Ni hao!

Sure. It is in the doc to do so. And it is simple to test.

In general, it seems a good practice to stick to the react packages. (react-hypescript worked for me but not react-native-hyperscript which was a badly maintained fork).

File: package.json (hunk 1 above)
Comment (MagicPoulp) [reply package.json]: yes. I fix it.

**Comment by MagicPoulp** [comment]
I updated the first commit with the changes you requested. You can merge the pull request if it suits you.

Playing with your code for my own project, I have decided to:
- move actions into new classes instead of having all possible actions in the props of top container. But I'm very into OO. I like that props have not too much in it and it reduces the coupling.
- I made the container into a new Component that will explicitly propagate the state variables to the sub-components. I found terribly interesting the forums discussing how much should one propagate from containers or connect the sub-components directly, and both have pros and cons.
https://stackoverflow.com/questions/34214208/why-redux-suggests-to-only-connect-to-top-level-components

In short, propagating has lower performance (without optimizing with the shouldUpdate...) but is more structured and more according to the old doc.

- I used eslint, and hyperscript.

Hyperscript is shorter and more programmable. This is how it looks like:
import h from "react-hyperscript";
h(Provider, { store: store }, h(App))

**Comment by MagicPoulp** [comment]
also:
I removed the BURK files
I much prefer Redux to mobx. With mobx, I can add KBs to my bundle and everything that happen is hidden.


I initalized the state to {} at the creation of the store.
EDIT: I was wrong, actually, reducers should explicitly initialize the state, as said here: https://stackoverflow.com/questions/36619093/why-do-i-get-reducer-returned-undefined-during-initialization-despite-pr

**Comment by MagicPoulp** [comment]
Is it good to be merged now? You can merge.

**Review by chentsulin — APPROVED** [review]

**Comment by chentsulin** [comment]
Merged. Thanks!
